subgraphs/hop-protocol/protocols/hop-protocol/src/L2/reward.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  RewardPaid,
+  Staked,
+  Withdrawn,
+} from "../../../../generated/HopL2Rewards/L2_Reward";
+import { Token } from "../../../../generated/schema";
+import { getUsdPricePerToken, getUsdPrice } from "../../../../src/prices/index";
+import { bigIntToBigDecimal } from "../../../../src/sdk/util/numbers";
+
+import {
+  updateRewardsPaid,
+  updateStaked,
+  updateWithdrawn,
+} from "../../../../src/sdk/util/rewards";
+import { conf } from "../../../../src/sdk/util/bridge";
+import { FOUR } from "../../../../src/sdk/util/constants";
+
+class Pricer implements TokenPricer {
+  getTokenPrice(token: Token): BigDecimal {
+    const price = getUsdPricePerToken(Address.fromBytes(token.id));
+    return price.usdPrice;
+  }
+
+  getAmountValueUSD(token: Token, amount: BigInt): BigDecimal {
+    const _amount = bigIntToBigDecimal(amount, token.decimals);
+    return getUsdPrice(Address.fromBytes(token.id), _amount);
+  }
+}
+
+class TokenInit implements TokenInitializer {
+  getTokenParams(address: Address): TokenParams {
+    const tokenConfig = NetworkConfigs.getTokenDetails(address.toHex());
+
+    if (tokenConfig.length != FOUR) {
+      log.error("Invalid tokenConfig length", []);
+    }
+
+    const name = tokenConfig[1];
+    const symbol = tokenConfig[0];
+    const decimals = BigInt.fromString(tokenConfig[2]).toI32();
+    return { name, symbol, decimals };
+  }
+}
+
+export function handleRewardsPaid(event: RewardPaid): void {
+  if (
+    !NetworkConfigs.getRewardTokenList().includes(event.address.toHexString())
+  ) {
+    log.error("Missing Config", []);
+    return;
+  }
+  const amount = event.params.reward;
+  const poolAddress = NetworkConfigs.getPoolAddressFromRewardTokenAddress(
+    event.address.toHexString()
+  );
+
+  const poolConfig = NetworkConfigs.getPoolDetails(poolAddress);
+  log.info("GNO RewardsPaid 1 --> poolAddress: {},", [poolAddress]);
+
+  const poolName = poolConfig[1];
+
+  const poolSymbol = poolConfig[0];
+  const hPoolName = poolConfig[2];
+
+  const sdk = SDK.initializeFromEvent(
+    conf,
+    new Pricer(),
+    new TokenInit(),
+    event
+  );
+  updateRewardsPaid(
+    sdk.Pools,
+    sdk.Pools,
+    sdk.Tokens,
+    poolName,
+    poolSymbol,
+    hPoolName,
+    poolAddress,
+    event,
+    amount
+  );
+}
+
+export function handleStaked(event: Staked): void {
+  if (
+    !NetworkConfigs.getRewardTokenList().includes(event.address.toHexString())
+  ) {
+    log.error("Missing Config", []);
+    return;
+  }
+  const amount = event.params.amount;
+
+  const poolAddress = NetworkConfigs.getPoolAddressFromRewardTokenAddress(
+    event.address.toHexString()
+  );
+  log.info("Staked --> emitter: {}, poolAddress: {}, amount: {}", [
+    event.address.toHexString(),
+    poolAddress,
+    amount.toString(),
+  ]);
+
+  const poolConfig = NetworkConfigs.getPoolDetails(poolAddress);
+  log.info("Staked 1 --> poolAddress: {},", [poolAddress]);
+
+  const poolName = poolConfig[1];
```

Comment:
error checking for array size in this file too

Reply:
I’ll fix that.